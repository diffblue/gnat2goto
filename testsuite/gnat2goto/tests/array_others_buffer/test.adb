--  Test based on StratoX s-bbcppr.adb code to detect unsupported function
with System; use System;
procedure Test is

   type Context_Id is range 0..100;
   SP_Process : constant Context_Id := 8;

   type Context_Buffer is array (Context_Id) of System.Address;
 
   procedure Initialize_Context
     (Buffer : not null access Context_Buffer;
      New_SP :  System.Address)
   is
   begin

     Buffer.all := (SP_process => New_SP, others => Null_Address);
   end Initialize_Context;

   Buff : access Context_Buffer;
   Addr : constant System.Address := Null_Address;
begin

   Initialize_Context (Buff, Addr);

   pragma Assert (Buff (8) = Null_Address);
 
end Test;
