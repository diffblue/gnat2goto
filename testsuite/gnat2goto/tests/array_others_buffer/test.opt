ALL XFAIL gnat2goto has unsupported others in aggregates
