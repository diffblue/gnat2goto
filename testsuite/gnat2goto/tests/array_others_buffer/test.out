Standard_Error from gnat2goto test:
test.adb:25:19: warning: "Buff" may be null

[test.assertion.1] line 25 Ada Check assertion: SUCCESS
[test.assertion.2] line 25 assertion Buff (8) = Null_Address: SUCCESS
[test.assertion.3] line 26 assertion Buff (0) = Null_Address: SUCCESS
[test.assertion.4] line 27 assertion Buff (100) = Null_Address: SUCCESS
[test.assertion.5] line 28 assertion Buff (50) = Null_Address: SUCCESS
[test.assertion.6] line 30 assertion Buff (99) /= Null_Address: FAILURE
VERIFICATION FAILED
